include/deal.II/grid/reference_cell.h

```diff
@@ -1119,25 +1119,29 @@ ReferenceCell::n_vertices() const
 inline unsigned int
 ReferenceCell::n_lines() const
 {
-  if (*this == ReferenceCells::Vertex)
-    return 0;
-  else if (*this == ReferenceCells::Line)
-    return 1;
-  else if (*this == ReferenceCells::Triangle)
-    return 3;
-  else if (*this == ReferenceCells::Quadrilateral)
-    return 4;
-  else if (*this == ReferenceCells::Tetrahedron)
-    return 6;
-  else if (*this == ReferenceCells::Pyramid)
-    return 7;
-  else if (*this == ReferenceCells::Wedge)
-    return 9;
-  else if (*this == ReferenceCells::Hexahedron)
-    return 12;
+  switch (this->kind)
+    {
+      case ReferenceCells::Vertex:
+        return 0;
+      case ReferenceCells::Line:
+        return 1;
+      case ReferenceCells::Triangle:
+        return 3;
+      case ReferenceCells::Quadrilateral:
+        return 4;
+      case ReferenceCells::Tetrahedron:
+        return 6;
+      case ReferenceCells::Pyramid:
+        return 7;
+      case ReferenceCells::Wedge:
+        return 9;
+      case ReferenceCells::Hexahedron:
+        return 12;
+      default:
+        Assert(false, ExcNotImplemented());
+    }
 
-  Assert(false, ExcNotImplemented());
-  return 0;
+  return numbers::invalid_unsigned_int;
```

Comment:
This was intentionally. See https://github.com/dealii/dealii/pull/14283.

Reply:
You're right - lets change that back.